src/actions/reportsActions.js

```diff
@@ -227,6 +245,36 @@ export const generateReportACreator = (
   };
 };
 
+export const generateReportToHardDriveACreator = (addList,pageList,saveAsFavorite,availableUntil) => dispatch =>  {
+  return new Promise((resolve, reject) => {
+    const teamsSheets = {};
+    for (let i = 0; i < addList.length; i++) {
+      teamsSheets[addList[i].name] = pageList[i].value;
+    }
+
+    let model = {
+      teamsSheets: teamsSheets,
+      saveAsFavorite: saveAsFavorite,
+      availableUntil: availableUntil
+    };
+
+    dispatch(setIsStarted(true, "Generowanie raportu"));
+      WebApi.reports.post
+        .report(model, false, false)
+        .then(response => {
+          dispatch(setIsStarted(false, ""));
+          dispatch(generateReport(true, []));
```

Comment:
- async dispatching, are u sure there is no need to do it synchronically ?

Reply:
https://stackoverflow.com/questions/43276291/is-store-dispatch-in-redux-synchronous-or-asynchronous

Reply:
dispatch as a method is synchronous but the whole method is handled async, so on:
we are running 2 dispatches together and we don't know when they will return a result.
(ale teraz to już nie wiem, pamiętam że jak robiłem tak to mi sie krzaczyło)

Reply:
![image](https://user-images.githubusercontent.com/16143176/48028157-0fe0fd00-e14b-11e8-8bd0-f4f29751fcca.png)
![image](https://user-images.githubusercontent.com/16143176/48028166-17a0a180-e14b-11e8-8e97-7214601b9481.png)